tests/eval_tests/tool_test_analysis.ipynb

```diff
@@ -0,0 +1,1394 @@
+    "        return\n",
+    "\n",
+    "    llama_client = LlamaStackClient(base_url=base_url)\n",
+    "\n",
+    "    # Define models to test\n",
+    "    # make sure they are available in your LLS server\n",
+    "    models = [\"llama3.2:1b\"]\n",
+    "\n",
+    "    client_tool_queries = os.path.join(os.getcwd(), \"queries/\", \"client_tool_queries.json\")\n",
+    "\n",
+    "    # Track statistics\n",
+    "    total_tests = 0\n",
+    "    successful_tests = 0\n",
+    "\n",
+    "    # Loop through models (outermost loop)\n",
+    "    for model in models:\n",
+    "        logger.info(f\"\\n=== Testing with model: {model} ===\\n\")\n",
+    "\n",
+    "        if client_tool_queries:\n",
+    "            queries = load_queries(client_tool_queries)\n",
+    "\n",
+    "            if not queries:\n",
+    "                logger.info(f\"No queries found in {client_tool_queries}\")\n",
+    "                continue\n",
+    "\n",
+    "            for query_obj in queries:\n",
+    "                total_tests += 1\n",
+    "                success = run_client_tool_test(model, query_obj, tools, llama_client, logger)\n",
+    "                if success:\n",
+    "                    successful_tests += 1\n",
+    "\n",
+    "    # Print summary\n",
+    "    logger.info(f\"\\n=== Test Summary ===\")\n",
+    "    logger.info(f\"Total tests: {total_tests}\")\n",
+    "    logger.info(f\"Successful tests: {successful_tests}\")\n",
+    "    logger.info(f\"Failed tests: {total_tests - successful_tests}\")\n",
+    "    if total_tests > 0:\n",
+    "        success_rate = (successful_tests / total_tests) * 100\n",
+    "        logger.info(f\"Success rate: {success_rate:.1f}%\")\n",
+    "\n",
+    "    # Generate plots\n",
+    "    logger.info(f\"\\n=== Generating plots ===\")\n",
+    "    utils.get_analysis_plots('./results/normal_client_tool_metrics_1B.csv')\n",
+    "\n",
+    "\n",
+    "main()\n"
+   ],
+   "id": "ca60c1aea5b97699"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "# Results\n",
+    "\n",
+    "![tool_calling_match_per_function_23_tools_1B.png.jpg](results/plots/tool_calling_match_per_function_23_tools_1B.png.jpg)\n",
+    "![tool_calling_match_per_function_1_tool_1B.jpg](results/plots/tool_calling_match_per_function_1_tool_1B.jpg)\n",
+    "\n",
+    "The results show that 1B param models, at least for llama family, are too small for tool calling.\n",
+    "\n",
+    "Another test was done where the model was **explicitly** told to use tool calling and given **only** the correct tool, in order to test whether the tool selection process was offloaded if 1B param models would work, however surprisingly it got 0 successful tool calls then."
+   ],
+   "id": "ba8bab8a04bd6532"
+  },
+  {
+   "metadata": {},
+   "cell_type": "markdown",
+   "source": [
+    "# Summary\n",
+    "\n",
+    "The following observations have been made using our eval set of 600 queries.\n",
```

Comment:
It would be a good idea to talk a bit amore this eval set at the top of the notebook.